integrations/airflow/marquez_airflow/dag.py

```diff
@@ -77,44 +74,47 @@ def create_dagrun(self, *args, **kwargs):
 
         return dagrun
 
-    def _register_dagrun(self, dagrun, execution_date, run_args):
+    # TODO: do we actually need to start tasks here or just register a hierarchy?
+    # Doing it other way would require to hook up to
+    # scheduler, where tasks are actually started
```

Comment:
We make the assumption that when a DAG run is created, it's tasks can be _safely_ marked as **started** as well. This will, as you note, help build the graph hierarchy. But, not all tasks start at the same time. So ideally, we'd only start the task right before it's scheduled for execution by a worker.

Reply:
Agreed. Although, sounds good for a future task.